packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -70,6 +72,30 @@ Future<Null> pubGet({
 
 final RegExp _analyzerWarning = new RegExp(r'^! \w+ [^ ]+ from path \.\./\.\./bin/cache/dart-sdk/lib/\w+$');
 
+/// The console environment key used by the pub tool.
+const String _pubEnvironmentKey = 'PUB_ENVIRONMENT';
+
+/// Returns the environment value that should be used when running pub.
+///
+/// Includes any existing environment variable, if one exists.
+String _getPubEnvironment() {
+  final List<String> values = <String>[];
+
+  final String existing = platform.environment[_pubEnvironmentKey];
+
+  if (existing != null && existing.isNotEmpty) {
+    values.add(existing);
```

Comment:
Does the existing values string need to be split around ':' and then rejoined below?